core/imgread/common.cpp

```diff
@@ -269,9 +269,7 @@ bool DiscSwap(u32 fileflags)
 
 void TermDrive()
 {
-	if (disc != NULL)
-		delete disc;
-
+	delete disc;
```

Comment:
I saw your exchange with skmp about this. Although this is correct, I prefer to be explicit that some memory may not have been allocated.